loom/test/test_infer.py

```diff
@@ -0,0 +1,31 @@
+import pandas as pd
+import numpy as np
+import json
+
+import loom.tasks
+
+def _learn_seeded_model(seed):
+    """Helper function to learn the exact same taxi model with different seeds"""
+    config = {'schedule': {'extra_passes': 1.0}, 'seed':seed}
+    name = 'seeding-test'
+    loom.tasks.ingest(name, 'synth-schema.json', 'synth.csv', debug=True)
+    loom.tasks.infer(name, sample_count=1, config=config, debug=True)
+    with loom.tasks.query(name) as server:
+        dependencies = server.relate(['a', 'b'])
+    return dependencies
+
+def test_seeding():
+    """Test if we can seed learning"""
+    schema = {'a':'nich', 'b': 'nich'}
+    with open('synth-schema.json', 'w') as outfile:
+        json.dump(schema, outfile)
+    df = pd.DataFrame({
+        'a':np.random.normal(0, 1, size=10),
+        'b':np.random.normal(0, 1, size=10)
+    })
+    df.to_csv('synth.csv', index=False)
+    dependencies_1 = _learn_seeded_model(42)
+    dependencies_2 = _learn_seeded_model(42)
+    assert dependencies_1 == dependencies_2, 'Setting the seeed did not work'
+    dependencies_3 = _learn_seeded_model(43)
+    assert dependencies_1 != dependencies_3, 'Auto-seeding overwrites our seed'
```

Comment:
Inequality of the dependence partition is neither sufficient nor necessary to conclude that the two seeds are different... perhaps we could try to confirm a single numeric parameter is not the same? (I don't think in general one can test from the end state whether the seeds are different, but a numeric parameter is less likely to be the same than a discrete dependence structure).

Reply:
see prev comment above.